storage/memory/peer_store.go

```diff
@@ -84,10 +84,12 @@ type peerStore struct {
 
 var _ storage.PeerStore = &peerStore{}
 
-func (s *peerStore) shardIndex(infoHash bittorrent.InfoHash, p bittorrent.Peer) uint32 {
-	idx := binary.BigEndian.Uint32(infoHash[:4]) % uint32(len(s.shards))
-	if len(p.IP) == net.IPv6len {
-		idx += idx + uint32(len(s.shards)/2)
+func (s *peerStore) shardIndex(infoHash bittorrent.InfoHash, v6 bool) uint32 {
+	// we actually have twice the amount of shards specified by the user:
```

Comment:
I've been trying my best not to ever write "we" or "our" in documentation.

"There are twice the amount of shards specified by the user, the first half is dedicated to IPv4 swarms and the second half is dedicated to IPv6 swarms"